.github/workflows/test-services.yml

```diff
@@ -0,0 +1,38 @@
+name: Services
+on:
+  pull_request:
+    types: [opened, edited, reopened, synchronize]
+
+jobs:
+  test-services:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup
+        uses: ./.github/actions/setup
+        with:
+          node-version: 16
+
+      - name: Service tests (triggered from local branch)
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        uses: ./.github/actions/service-tests
+        with:
+          github-token: '${{ secrets.GH_PAT }}'
+          librariesio-tokens: '${{ secrets.SERVICETESTS_LIBRARIESIO_TOKENS }}'
+          obs-user: '${{ secrets.SERVICETESTS_OBS_USER }}'
+          obs-pass: '${{ secrets.SERVICETESTS_OBS_PASS }}'
+          sl-insight-user-uuid: '${{ secrets.SERVICETESTS_SL_INSIGHT_USER_UUID }}'
+          sl-insight-api-token: '${{ secrets.SERVICETESTS_SL_INSIGHT_API_TOKEN }}'
+          twitch-client-id: '${{ secrets.SERVICETESTS_TWITCH_CLIENT_ID }}'
+          twitch-client-secret: '${{ secrets.SERVICETESTS_TWITCH_CLIENT_SECRET }}'
+          wheelmap-token: '${{ secrets.SERVICETESTS_WHEELMAP_TOKEN }}'
+          youtube-api-key: '${{ secrets.SERVICETESTS_YOUTUBE_API_KEY }}'
+
+      - name: Service tests (triggered from fork)
+        if: github.event.pull_request.head.repo.full_name != github.repository
+        uses: ./.github/actions/service-tests
+        with:
+          github-token: '${{ secrets.GITHUB_TOKEN }}'
```

Comment:
If we submit a PR from a branch on this repo (Anyone with push access can do this. Also importantly, this applies to dependabot PRs - they're subimtted from a local branch, not a fork), all of the service tests can run using credentials which I have configured as secrets.

If we recieve a PR from a fork (like this one) tests for the following services will fail due to lack of credentials, or be flaky due to reduced rate limit:
- Libraries.io
- OBS
- Symfony Insight
- Twitch
- Wheelmap
- Youtube
- GitHub gists only, but not the rest of GitHub (more on this further down)

All other service tests that do not require auth will run fine in either case.
As discussed on the call, this means this are a small handful of services where if an external contributor wants to submit a PR changing them, we'll have to verify the tests locally and then edit the PR title before merging.
Seperately, I will have a look at finding a way to run tests for these services in another context.

You can see an example of service tests running with the required credentials on a PR from a local branch at https://github.com/badges/shields/actions/runs/3923463559/jobs/6707091330

Reply:
here's an equivalent build trying to run the same service tests from a fork
https://github.com/badges/shields/actions/runs/3923735407